packages/ecs/index.ts

```diff
@@ -0,0 +1,107 @@
+import {Construct} from "constructs";
+import {DEFAULT_ARCHITECTURE, DEFAULT_MAIN_VPC_NAME, DEFAULT_MEMORY_GB, DEFAULT_VCPUS} from "./config";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as ecs from "aws-cdk-lib/aws-ecs";
+import {ContainerInsights} from "aws-cdk-lib/aws-ecs";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as ecrAssets from "aws-cdk-lib/aws-ecr-assets";
+import {RetentionDays} from "aws-cdk-lib/aws-logs";
+
+// Memory and CPU limits for Fargate tasks
+/*
+256 (.25 vCPU) - Available memory values: 512 (0.5 GB), 1024 (1 GB), 2048 (2 GB)
+
+512 (.5 vCPU) - Available memory values: 1024 (1 GB), 2048 (2 GB), 3072 (3 GB), 4096 (4 GB)
+
+1024 (1 vCPU) - Available memory values: 2048 (2 GB), 3072 (3 GB), 4096 (4 GB), 5120 (5 GB), 6144 (6 GB), 7168 (7 GB), 8192 (8 GB)
+
+2048 (2 vCPU) - Available memory values: Between 4096 (4 GB) and 16384 (16 GB) in increments of 1024 (1 GB)
+
+4096 (4 vCPU) - Available memory values: Between 8192 (8 GB) and 30720 (30 GB) in increments of 1024 (1 GB)
+
+8192 (8 vCPU) - Available memory values: Between 16384 (16 GB) and 61440 (60 GB) in increments of 4096 (4 GB)
+
+16384 (16 vCPU) - Available memory values: Between 32768 (32 GB) and 122880 (120 GB) in increments of 8192 (8 GB)
+*/
+
+export type Architecture = 'X86_64' | 'ARM64'
+
+export const CPU_ARCHITECTURE_MAP: Record<Architecture, ecs.CpuArchitecture> = {
+    ['X86_64']: ecs.CpuArchitecture.X86_64,
+    ['ARM64']: ecs.CpuArchitecture.ARM64
+}
+
+export const LAMBDA_ARCHITECTURE_MAP: Record<Architecture, lambda.Architecture> = {
+    ['X86_64']: lambda.Architecture.X86_64,
+    ['ARM64']: lambda.Architecture.ARM_64
+}
+
+export interface FargateEcsTaskConstructProps {
```

Comment:
If you use typedoc convention for the props here, the description will then appear in the markdown docs and when you hover the props in the IDE (at the main stack when importing this package)

e.g. https://github.com/OrcaBus/platform-cdk-constructs/blob/main/packages/deployment-stack-pipeline/pipeline.ts#L69-L122

Reply:
Also since the build is using `jsii` before publishing to npm. The jsii required the exported interface to only be `readonly`

Reply:
Ah awesome, done for #5, #6 and #7